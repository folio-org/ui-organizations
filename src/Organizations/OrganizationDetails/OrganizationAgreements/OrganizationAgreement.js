import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import {
  Row,
  Col,
  KeyValue,
  NoValue,
} from '@folio/stripes/components';

const OrganizationAgreement = ({ name, discount, referenceUrl, notes }) => {
  return (
    <Row>
      <Col xs={3}>
        <KeyValue
          label={<FormattedMessage id="ui-organizations.agreement.name" />}
          value={name}
        />
      </Col>
      <Col xs={3}>
        <KeyValue
          label={<FormattedMessage id="ui-organizations.agreement.discount" />}
          value={discount !== undefined ? `${discount}%` : <NoValue />}
        />
      </Col>
      <Col xs={3}>
        <KeyValue
          label={<FormattedMessage id="ui-organizations.agreement.referenceUrl" />}
          value={referenceUrl || <NoValue />}
        />
      </Col>
      <Col xs={12}>
        <KeyValue
          label={<FormattedMessage id="ui-organizations.agreement.notes" />}
          value={notes || <NoValue />}
        />
      </Col>
    </Row>
  );
};

OrganizationAgreement.propTypes = {
  name: PropTypes.string,
  discount: PropTypes.number,
  referenceUrl: PropTypes.string,
  notes: PropTypes.string,
};

export default OrganizationAgreement;
