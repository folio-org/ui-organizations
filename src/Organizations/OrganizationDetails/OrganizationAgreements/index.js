export { default as OrganizationAgreement } from './OrganizationAgreement';
export { default as OrganizationAgreements } from './OrganizationAgreements';
