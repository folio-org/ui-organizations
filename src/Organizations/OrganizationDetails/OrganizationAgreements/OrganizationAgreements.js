import React from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import OrganizationAgreement from './OrganizationAgreement';

const OrganizationAgreements = ({ agreements }) => {
  if (!agreements.length) {
    return (
      <p>
        <FormattedMessage id="ui-organizations.agreement.noAgreementsAvailable" />
      </p>
    );
  }

  return (
    <div data-test-agreements>
      {
        agreements.map((agreement, idx) => (
          <React.Fragment key={idx}>
            <OrganizationAgreement
              name={agreement.name}
              discount={agreement.discount}
              referenceUrl={agreement.referenceUrl}
              notes={agreement.notes}
            />

            {idx !== agreements.length && <hr />}
          </React.Fragment>
        ))
      }
    </div>
  );
};

OrganizationAgreements.propTypes = {
  agreements: PropTypes.arrayOf(PropTypes.object),
};

OrganizationAgreements.defaultProps = {
  agreements: [],
};

export default OrganizationAgreements;
