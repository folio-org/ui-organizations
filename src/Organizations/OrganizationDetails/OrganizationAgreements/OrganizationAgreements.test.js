import React from 'react';
import { render } from '@testing-library/react';

import OrganizationAgreements from './OrganizationAgreements';

const agreements = [{
  name: 'Amazon V',
  discount: 10,
  referenceUrl: 'test.com',
  notes: 'Default',
}];
const renderOrganizationAccounts = () => render(
  <OrganizationAgreements
    agreements={agreements}
  />,
);

describe('OrganizationAgreements', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure', () => {
    const { asFragment } = renderOrganizationAccounts();

    expect(asFragment()).toMatchSnapshot();
  });
});
