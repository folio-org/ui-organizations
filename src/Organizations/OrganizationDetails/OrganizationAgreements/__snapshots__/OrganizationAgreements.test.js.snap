// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationAgreements should render correct structure 1`] = `
<DocumentFragment>
  <div
    data-test-agreements="true"
  >
    <div
      class="row"
    >
      <div
        class="col-xs-3"
      >
        <div
          class="kvRoot"
        >
          <div
            class="kvLabel"
          >
            ui-organizations.agreement.name
          </div>
          <div
            class="kvValue"
            data-test-kv-value="true"
          >
            Amazon V
          </div>
        </div>
      </div>
      <div
        class="col-xs-3"
      >
        <div
          class="kvRoot"
        >
          <div
            class="kvLabel"
          >
            ui-organizations.agreement.discount
          </div>
          <div
            class="kvValue"
            data-test-kv-value="true"
          >
            10%
          </div>
        </div>
      </div>
      <div
        class="col-xs-3"
      >
        <div
          class="kvRoot"
        >
          <div
            class="kvLabel"
          >
            ui-organizations.agreement.referenceUrl
          </div>
          <div
            class="kvValue"
            data-test-kv-value="true"
          >
            test.com
          </div>
        </div>
      </div>
      <div
        class="col-xs-12"
      >
        <div
          class="kvRoot"
        >
          <div
            class="kvLabel"
          >
            ui-organizations.agreement.notes
          </div>
          <div
            class="kvValue"
            data-test-kv-value="true"
          >
            Default
          </div>
        </div>
      </div>
    </div>
    <hr />
  </div>
</DocumentFragment>
`;
